src/snowflake/snowpark/session.py

```diff
@@ -999,6 +1003,84 @@ def table_function(
+
+        References: `Snowflake Generator function <https://docs.snowflake.com/en/sql-reference/functions/generator.html>`_.
+
+        Args:
+            columns: List of data generation function that work in tandem with generator table function.
+            rowcount: Resulting table with contain ``rowcount`` rows if only this argument is specified. Defaults to 0.
+            timelimit: The query runs for ``timelimit`` seconds, generating as many rows as possible within the time frame. The
+                exact row count depends on the system speed. Defaults to 0.
+
+        Example 1
+            >>> from snowflake.snowpark.functions import seq1, seq8, uniform
+            >>> df = session.generator(seq1(1).as_("sequence one"), uniform(1, 10, 2).as_("uniform"), rowcount=3)
+            >>> df.show()
+            ------------------------------
+            |"sequence one"  |"UNIFORM"  |
+            ------------------------------
+            |0               |3          |
+            |1               |3          |
+            |2               |3          |
+            ------------------------------
+            <BLANKLINE>
+
+        Example 2
+            >>> df = session.generator(seq8(0), uniform(1, 10, 2), timelimit=1).order_by(seq8(0)).limit(3)
+            >>> df.show()
+            -----------------------------------
+            |"SEQ8(0)"  |"UNIFORM(1, 10, 2)"  |
+            -----------------------------------
+            |0          |3                    |
+            |1          |3                    |
+            |2          |3                    |
+            -----------------------------------
+            <BLANKLINE>
+
+        Note:
+            - When both ``rowcount`` and ``timelimit`` are specified, then:
+
+                + if the ``rowcount`` is reached before the ``timelimit``, the resulting table with contain ``rowcount`` rows.
+                + if the ``timelimit`` is reached before the ``rowcount``, the table will contain as many rows generated within this time.
+
+            - If both ``rowcount`` and ``timelimit`` are not specified, 0 rows will be generated.
+
+        Returns:
+            A new :class:`DataFrame` with data from calling the generator table function.
+        """
+        if not columns:
+            raise ValueError("Columns cannot be empty for generator table function")
+        named_args = {}
+        if rowcount != 0:
+            named_args["rowcount"] = lit(rowcount)._expression
+        if timelimit != 0:
+            named_args["timelimit"] = lit(timelimit)._expression
+
+        operators = [self._analyzer.analyze(col._expression) for col in columns]
+        func_expr = GeneratorTableFunction(args=named_args, operators=operators)
+
+        if self.sql_simplifier_enabled:
+            d = DataFrame(
+                self,
+                SelectStatement(
+                    from_=SelectTableFunction(
+                        func_expr=func_expr, analyzer=self._analyzer
```

Comment:
not related to this PR: @sfc-gh-yixie we don't need to pass in analyzer twice, right? We can fix it after GA.

Reply:
Unfortunately right now we need to have it twice.